### PR TITLE
CustomScrollView

### PR DESCRIPTION
**Pull request by abarth**
Also, use CustomScrollView in Shrine and fix a bug with one-line grids
not painting properly due to their reporiting zero paintExtent.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: examples/flutter_gallery/lib/demo/shrine/shrine_order.dart [hunk 1]
```diff
@@ -174,39 +174,47 @@ class _OrderPageState extends State<OrderPage> {
...
+              delegate: new SliverChildListDelegate(
+                config.products
+                  .where((Product product) => product != config.order.product)
+                  .map((Product product) {
+                    return new Card(
+                      elevation: 1,
+                      child: new Image.asset(
+                        product.imageAsset,
+                        fit: ImageFit.contain,
+                    ),
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/shrine/shrine_order.dart]: looks like we got the alignment of the parentheses off a bit here

File: packages/flutter/lib/src/rendering/sliver_grid.dart [hunk 2]
```diff
@@ -61,6 +65,16 @@ class SliverGridGeometry {
...
     );
   }
+
+  @override
+  String toString() {
+    return 'SliverGridGeometry('
+      'scrollOffset: $scrollOffset, '
+      'crossAxisOffset: $crossAxisOffset, '
+      'mainAxisExtent: $mainAxisExtent, '
+      'crossAxisExtent=$crossAxisExtent'
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/sliver_grid.dart]: s/=/:/ unless you have some convention i'm not aware of :-)

File: packages/flutter/test/widgets/grid_view_test.dart [hunk 3]
```diff
@@ -304,6 +305,29 @@ void main() {
     expect(find.text('4'), findsNothing);
   });
 
+  testWidgets('One-line GridView paings', (WidgetTester tester) async {
```
Comment (Hixie) [inline packages/flutter/test/widgets/grid_view_test.dart]: paings?

**Comment by Hixie** [comment]
LGTM

Thanks.

File: examples/flutter_gallery/lib/demo/shrine/shrine_order.dart (hunk 1 above)
Comment (abarth) [inline examples/flutter_gallery/lib/demo/shrine/shrine_order.dart]: Fixed.

File: packages/flutter/lib/src/rendering/sliver_grid.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/rendering/sliver_grid.dart]: Ugg.  They started out as `=` and I forgot to convert that one.

**Pull request merged**